TYPING: `_64Bit` reported as incompatible with `NBitBase`

### Describe the issue:

`mypy` reports an error in the following code. Comparing `ndarray[Any, dtype[signedinteger[_64Bit]]]` and `ndarray[Any, dtype[signedinteger[NBitBase]]]`, the only difference seems to be `_64Bit` vs. `NBitBase` - although the former is a subclass of the latter:

https://github.com/numpy/numpy/blob/a277f6210739c11028f281b8495faf7da298dbef/numpy/_typing/__init__.py#L67-L80

### Reproduce the code example:

```python
import numpy as np
import numpy.typing as npt

def fun(array: npt.NDArray[np.bool_]) -> npt.NDArray[np.signedinteger[npt.NBitBase]]:
    return np.flatnonzero(array)
```

### Error message:

```shell
C:\Code\project>mypy bug.py
bug.py:5: error: Incompatible return value type (got "ndarray[Any, dtype[signedinteger[_64Bit]]]", expected "ndarray[Any, dtype[signedinteger[NBitBase]]]")  [return-value]
Found 1 error in 1 file (checked 1 source file)
```

### Runtime information:

1.25.2
3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)]

### Context for the issue:

I had to choose between reporting this at `mypy` or here first. Hope I made a decent choice, but I am happy to report elsewhere as well.

--- a/numpy/typing/__init__.py
+++ b/numpy/typing/__init__.py
@@ -93,7 +93,7 @@ Please see : :ref:`Data type objects <arrays.dtypes>`
 Number precision
 ~~~~~~~~~~~~~~~~
 
-The precision of `numpy.number` subclasses is treated as a covariant generic
+The precision of `numpy.number` subclasses is treated as a invariant generic
 parameter (see :class:`~NBitBase`), simplifying the annotating of processes
 involving precision-based casting.
 

